src/StoreApi/Routes/Checkout.php

```diff
@@ -264,7 +264,13 @@ protected function get_route_post_response( WP_REST_Request $request ) {
 		/**
 		 * Process the payment and return the results.
 		 */
-		$payment_result = $this->order->needs_payment() ? $this->process_payment( $request ) : $this->process_without_payment( $request );
+		$payment_result = new PaymentResult();
```

Comment:
Passing this through to the private methods to make the payment methods clearer.